hardware_integration/src/vector_asc_logger.cpp

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+** @file       vector_asc_logger.cpp
+** @author     Adrian Del Grosso
+** @copyright  The Open-Agriculture Developers
+*******************************************************************************/
+#include "isobus/hardware_integration/vector_asc_logger.hpp"
+
+#include "isobus/isobus/can_stack_logger.hpp"
+#include "isobus/utility/system_timing.hpp"
+#include "isobus/utility/to_string.hpp"
+
+#include <cmath>
+#include <ctime>
+#include <iomanip>
+
+namespace isobus
+{
+	VectorASCLogger::VectorASCLogger()
+	{
+		std::time_t currentTime = std::time(0);
+		std::tm *now = std::localtime(&currentTime);
+		std::string defaultFileName = "AgISOStackLog_" +
+		  isobus::to_string(now->tm_year + 1900) +
+		  "_" +
+		  isobus::to_string(now->tm_mon + 1) +
+		  "_" +
+		  isobus::to_string(now->tm_mday) +
+		  "_" +
+		  isobus::to_string(now->tm_hour) +
+		  "_" +
+		  isobus::to_string(now->tm_min) +
+		  "_" +
+		  isobus::to_string(now->tm_sec) +
+		  ".asc";
+		openFile(defaultFileName, constructHeaderTime(currentTime));
+	}
+
+	VectorASCLogger::VectorASCLogger(std::string filename)
+	{
+		std::time_t currentTime = std::time(0);
+
+		openFile(filename, constructHeaderTime(currentTime));
+	}
+
+	VectorASCLogger::~VectorASCLogger()
+	{
+		if (logFileStream)
+		{
+			logFileStream.close();
+		}
+	}
```

Comment:
The 2 event listeners should be de-registered here. The automatic disposal has been removed some time ago because it was giving more confusion than it was worth.

Reply:
Added! 👍